src/components/AllComponents.js

```diff
@@ -0,0 +1,539 @@
+import React, { useEffect, useState } from "react";
+import SingleCard from "./SingleCard";
+import SuggestToWeb from "./SuggestToWeb";
+import { handleRarityChange } from "../util/helpers";
+
+import { useQuery, useLazyQuery } from "@apollo/client";
+import { QUERY_CHARACTERS, QUERY_ONECHARACTER, GET_USERDATA, GET_USERCHARACTERSBYID } from "../util/queries";
+import CardDetails from "./CardDetails";
+
+import Auth from "../util/auth";
+
+function AllComponents() {
+  const categorySelect = document.getElementById("categories");
+
+  const [search, setSearch] = useState("");
```

Comment:
We've spent awhile on redoing the SearchForm, so I won't cover anything we already changed in the MD-Search-Optimization branch